cmd/share.go

```diff
@@ -0,0 +1,98 @@
+package cmd
+
+import (
+	"fmt"
+	"kool-dev/kool/cmd/builder"
+	"kool-dev/kool/environment"
+	"regexp"
+
+	"github.com/spf13/cobra"
+)
+
+// KoolShareFlags holds the flags for the kool stop command
+type KoolShareFlags struct {
+	Service   string
+	Subdomain string
+}
+
+// KoolShare holds handlers and functions to implement the stop command logic
+type KoolShare struct {
+	DefaultKoolService
+	Flags *KoolShareFlags
+
+	env environment.EnvStorage
+
+	status *KoolStatus
+	share  builder.Command
+}
+
+func init() {
+	var (
+		share    = NewKoolShare()
+		shareCmd = NewShareCommand(share)
+	)
+
+	rootCmd.AddCommand(shareCmd)
+}
+
+// NewKoolShare creates a new handler for sharing local environment with default dependencies
+func NewKoolShare() *KoolShare {
+	defaultKoolService := newDefaultKoolService()
+	return &KoolShare{
+		*defaultKoolService,
+		&KoolShareFlags{"app", ""},
+		environment.NewEnvStorage(),
+		NewKoolStatus(),
+		builder.NewCommand("docker", "run", "--rm", "--init"),
+	}
+}
+
+// validSubdomain runs the stop logic with incoming arguments.
+func (s *KoolShare) validSubdomain(subdomain string) bool {
+	return regexp.MustCompile("^[a-z]+[a-z0-9]+$").MatchString(subdomain)
```

Comment:
why this logic / validation? can't have _ or -?

Reply:
yeah this needs improvement, I will get thee RFC regex